cmd/virt-launcher/virt-launcher.go

```diff
@@ -335,6 +338,20 @@ func waitForFinalNotify(deleteNotificationSent chan watch.Event,
 	}
 }
 
+func runDownwardMetricsVirtioServer(ctx context.Context) error {
+	if _, ok := os.LookupEnv(services.ENV_VAR_VIRT_LAUNCHER_DOWNWARDMETRICS_SERVER); ok {
+		nodeName, ok := os.LookupEnv(services.ENV_VAR_NODE_NAME)
+		if !ok {
+			return errors.New(services.ENV_VAR_NODE_NAME + " environment variable not found")
+		}
+
+		if err := virtioserial.RunDownwardMetricsVirtioServer(ctx, nodeName, downwardmetrics.DownwardMetricsChannelSocket); err != nil {
```

Comment:
Can we add a unittest to make sure the behaviour is correct when we fail to start the server (due to an invalid socket path or anything else?).

and we should probably also test the context cancellation (below) too

Reply:
do we need to verify that downwardmetrics.DownwardMetricsChannelSocket already exists?

Reply:
I don't think so, the server can start before the socket file exist, so I think the safest option it to leave it to the connection code that will retry in that case (or if the socket exist but is not in the listen state yet)

Reply:
The context cancellation is already tested. I've added another one in case the `signalStopChannel` is called.